runtime/ops/indexing.cc

```diff
@@ -162,5 +163,20 @@ chainerx::Array WhereOp::RunImpl(ChxVMState* st, chainerx::Array const& conditio
     return chainerx::Where(condition, x, y);
 }
 
+chainerx::Array NonZeroOp::RunImpl(ChxVMState* st, const chainerx::Array& x_) {
+    chainerx::Array x = chainerx::AsContiguous(x_.AsType(chainerx::Dtype::kBool));
+    const int64_t rank = x.shape().size();
+    std::vector<int64_t> result;
+    chainerx::IndexIterator<chainerx::kDynamicNdim> idx_it(x.shape().data(), rank, x.shape().GetTotalSize(), 0, 1);
+    const bool* x_start = reinterpret_cast<const bool*>(x.raw_data());
```

Comment:
I think it's better to `CHECK(IsNativeDevice(&x.device()))` or move the array to the native device